src/heatmap.ts

```diff
@@ -0,0 +1,761 @@
+      ctx.blendFunc(ctx.ONE, ctx.ONE_MINUS_SRC_ALPHA);
+      ctx.depthMask(true);
+      layer.setAttribute("height", (height * this.ratio).toString());
+      layer.setAttribute("width", (width * this.ratio).toString());
+      layer.style.height = `${height}px`;
+      layer.style.width = `${width}px`;
+      layer.style.position = "absolute";
+      res.appendChild(layer);
+
+      this.ctx = ctx;
+      this.width = width;
+      this.height = height;
+      this.imageConfig = null;
+      this.configMin = null;
+      this.configMax = null;
+      this.hearmapExData = {};
+      this.layer = layer;
+      this.dom = res;
+      this.gradShadOP = createGradiantShader(this.ctx, GradShader);
+      this.colorShadOP = createColorShader(this.ctx, ColorShader);
+      this.imageShaOP = createImageShader(this.ctx, ImageShader);
+      this.fbTexObj = ctx.createTexture()!;
+      this.fbo = ctx.createFramebuffer()!;
+
+      if (!isNullUndefined(config.size)) {
+        this.setSize(config.size);
+      } else {
+        this.size = 20.0;
+      }
+
+      if (!isNullUndefined(config.max)) {
+        this.setMax(config.max);
+      } else {
+        this.configMax = null;
+      }
+
+      if (!isNullUndefined(config.min)) {
+        this.setMin(config.min);
+      } else {
+        this.configMin = null;
+      }
+
+      if (!isNullUndefined(config.intensity)) {
+        this.setIntensity(config.intensity);
+      } else {
+        this.intensity = 1.0;
+      }
+
+      if (!isNullUndefined(config.translate)) {
+        this.setTranslate(config.translate);
+      } else {
+        this.translate = [0, 0];
+      }
+
+      if (!isNullUndefined(config.zoom)) {
+        this.setZoom(config.zoom);
+      } else {
+        this.zoom = 1.0;
+      }
+
+      if (!isNullUndefined(config.angle)) {
+        this.setRotationAngle(config.angle);
+      } else {
+        this.angle = 0.0;
+      }
+
+      if (!isNullUndefined(config.opacity)) {
+        this.setOpacity(config.opacity);
+      } else {
+        this.opacity = 1.0;
+      }
+
+      this.gradient = gradientMapper(config.gradient);
+
+      if (config.backgroundImage && config.backgroundImage.url) {
+        this.setBackgroundImage(config.backgroundImage);
+      }
+
+      this.heatmapData = [];
+
+      this.ctx.viewport(0, 0, this.ctx.canvas.width, this.ctx.canvas.height);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  resize() {
+    const height = this.dom.clientHeight;
+    const width = this.dom.clientWidth;
+    this.layer.setAttribute("height", (height * this.ratio).toString());
+    this.layer.setAttribute("width", (width * this.ratio).toString());
+    this.layer.style.height = `${height}px`;
+    this.layer.style.width = `${width}px`;
+    this.width = width;
+    this.height = height;
+    this.ctx!.viewport(0, 0, this.ctx!.canvas.width, this.ctx!.canvas.height);
+    /* Perform update */
+    // TODO ISSUE
+    // this.render(this.hearmapExData);
```

Comment:
Here we are calling render function but that function takes no arguments, so commented the code for now, instead calling that function with no arguments. Wanted a clarification on this.

Reply:
render call is very much needed here.. as it re-renders the canvas based on new viewport dimension. Please remove the arguments here.